src/protocols/coda_transition_frontier.ml

```diff
@@ -190,14 +193,18 @@ module type Transition_frontier_intf = sig
 
   val iter : t -> f:(Breadcrumb.t -> unit) -> unit
 
-  val add_breadcrumb_exn : t -> Breadcrumb.t -> unit
+  val add_breadcrumb_exn : t -> Breadcrumb.t -> unit Deferred.t
```

Comment:
we very deliberately made `add_breadcrumb_exn` synchronous -- it's important to the correctness of the transition frontier system. We'll either have to revisit that design or we'll be forced to buffer the events (and my broadcast_pipe support buffered mode as well for this use-case). 

I wish we didn't need to ensure all the events were seen by listeners, but ya'll have already convinced me we do need that, so we'd need to do crash on overflow buffered mode probably.

cc @nholland94 any other ideas?

Reply:
Another idea is to bubble up the extensions via a return of this function (much like we did in transition_frontier0) and push them into their respective pipes with the pushback deferred where it's safer.

Reply:
As discussed in person, we'll bite the async bullet here, and as you've changed it should be fine :+1: